### PR TITLE
Added gradual change detection

### PR DESCRIPTION
**Pull request by calum-bird**

### CODE REVIEW

**Comment by calum-bird** [comment]
Alright the TLDR here is I've got no idea how the changes I made are causing this test to fail. Seems a bit peculiar.

**Review by Coolroo — APPROVED** [review]
LGTM!

File: src/autodoc/changeDetection.ts [hunk 1]
```diff
@@ -230,21 +247,34 @@ export class ChangeDetectionService {
...
+
+          if (newLevDistance > this.levenshtein_update_threshold) {
+            // If beyond the threshold, add to updated list
+            returnObj.updated.push(fundWithLevenshteinDistance);
           }
         }
         //If the function was deleted
         else {
           returnObj.deleted.push(functionPair["old"]);
+          returnObj.all.push(functionPair["old"]);
```
Comment (Coolroo) [inline src/autodoc/changeDetection.ts]: Are we persisting old functions? This probably shouldn't happen

**Review by Coolroo — CHANGES_REQUESTED** [review]

**Review by Coolroo — APPROVED** [review]
LGTM!

**Pull request merged**